tests/panda/test_hdf_panda.py

```diff
@@ -0,0 +1,235 @@
+    block_a.test_capture = SignalR(backend=SimSignalBackend(Capture, source="block_a"))
+    block_b.test_capture = SignalR(backend=SimSignalBackend(Capture, source="block_b"))
+
+    setattr(sim_hdf_panda, "block_a", block_a)
+    setattr(sim_hdf_panda, "block_b", block_b)
+    await sim_hdf_panda.connect(sim=True)
+    set_sim_value(block_a.test_capture, Capture.Min)
+    set_sim_value(block_b.test_capture, Capture.Diff)
+
+    yield sim_hdf_panda
+
+
+async def test_hdf_panda_passes_blocks_to_controller(sim_hdf_panda: HDFPanda):
+    assert hasattr(sim_hdf_panda.controller, "pcap")
+    assert sim_hdf_panda.controller.pcap is sim_hdf_panda.pcap
+
+
+async def test_hdf_panda_hardware_triggered_flyable(
+    RE: RunEngine,
+    sim_hdf_panda,
+):
+    names = []
+    docs = []
+
+    def append_and_print(name, doc):
+        names.append(name)
+        docs.append(doc)
+
+    RE.subscribe(append_and_print)
+
+    shutter_time = 0.004
+    exposure = 1
+
+    trigger_logic = StaticSeqTableTriggerLogic(sim_hdf_panda.seq[1])
+    flyer = HardwareTriggeredFlyable(trigger_logic, [], name="flyer")
+
+    def flying_plan():
+        yield from bps.stage_all(sim_hdf_panda, flyer)
+
+        yield from prepare_static_seq_table_flyer_and_detectors_with_same_trigger(
+            flyer,
+            [sim_hdf_panda],
+            num=1,
+            width=exposure,
+            deadtime=sim_hdf_panda.controller.get_deadtime(1),
+            shutter_time=shutter_time,
+        )
+        # sim_hdf_panda.controller.disarm.assert_called_once  # type: ignore
+
+        yield from bps.open_run()
+        yield from bps.declare_stream(sim_hdf_panda, name="main_stream", collect=True)
+
+        set_sim_value(flyer.trigger_logic.seq.active, 1)
+
+        yield from bps.kickoff(flyer, wait=True)
+        yield from bps.kickoff(sim_hdf_panda)
+
+        yield from bps.complete(flyer, wait=False, group="complete")
+        yield from bps.complete(sim_hdf_panda, wait=False, group="complete")
+
+        # Manually incremenet the index as if a frame was taken
+        set_sim_value(
+            sim_hdf_panda.data.num_captured,
+            sim_hdf_panda.data.num_captured._backend._value + 1,
+        )
+
+        set_sim_value(flyer.trigger_logic.seq.active, 0)
+
+        done = False
+        while not done:
+            try:
+                yield from bps.wait(group="complete", timeout=0.5)
+            except TimeoutError:
+                pass
+            else:
+                done = True
+            yield from bps.collect(
+                sim_hdf_panda,
+                return_payload=False,
+                name="main_stream",
+            )
+        yield from bps.wait(group="complete")
+        yield from bps.close_run()
+
+        yield from bps.unstage_all(flyer, sim_hdf_panda)
+        # assert sim_hdf_panda.controller.disarm.called  # type: ignore
+
+    # fly scan
+    RE(flying_plan())
+
+    assert names == [
```

Comment:
Please can you use the pattern for https://github.com/bluesky/ophyd-async/issues/206. You can copy the `assert_emitted` function into this file until #199 is done